### DISCUSSION TITLE
Is there any elegant way to stop/kill specific scheduled service?

### DISCUSSION THREAD

**Discussion by Studnia1** [jamesmh/coravel#289] [Q&A]
Hello, i'm using coravel for my private ASP project and I wonder if there's any elegant way to kill/stop sepcific scheduled service?

**Comment by jamesmh** [comment]
This method exists on the scheduler: `public bool TryUnschedule(string uniqueIndentifier)`. You'll have to schedule your invocables using either `AssignUniqueIndentifier(string uniqueIdentifier)` or `PreventOverlapping(string x)`.

```csharp
scheduler.Schedule<MyInvocable>()
.AssignedUniqueIdentifier(myId);
```

Keep track of your ids, and once you want to cancel do:

```csharp
var success = schedule.TryUnschedule(id);
```

These methods aren't in the docs but should be added. Let me know if that helps.

**Reply by Studnia1** [reply]
![image](https://user-images.githubusercontent.com/49270452/163579719-26f82f27-6d91-4c89-95ff-59030499286b.png)
That is on all methods mentioned above, im using coravel 4.1.1 from nuget packet manager, Am I doing something wrong?

**Reply by siganberg** [reply]
I was looking for the TryUnschedule in `IScheduler` but actually found it in the concrete class. Any reason why it's not the interface?

**Comment by jamesmh** [comment]
Prevent overlapping should be available. Take a look at this comment: https://github.com/jamesmh/coravel/issues/77#issuecomment-460785919
